### PR TITLE
Some more fixes

### PR DESCRIPTION
**Pull request by simonduq**

### CODE REVIEW

**Comment by atiselsts** [comment]
Wow, that's a lot, thanks.

I don't feel good about the functional changes in Zoul's config, since I don't know the Orion board, so can't review them. And I would strongly prefer to keep this PR free of any (nontrivial) changes in C code. Can you submit the Zoul's patches as a separate PR to Contiki-NG so that George or others can review and merge them?

**Comment by simonduq** [comment]
OK, will look into this at a later point. Some of the commits should be taken directly here (those adding `PLATFORMS_ONLY` and such flags)

File: tests/compile-all/build.sh [hunk 1]
```diff
@@ -120,7 +120,7 @@ do
             example_dir=`dirname "$example"`
 
             # Build the goal
-            $LOG_INFO make -C "$example_dir" -j TARGET=$platform BOARD=$board $GOAL
+            $LOG_INFO make -C "$example_dir" -j TARGET=$platform BOARD=$board clean $GOAL
```
Comment (atiselsts) [inline tests/compile-all/build.sh]: This one just adds the "clean" to the log. I'm assuming that you also meant to add it to the compile command - will do this change myself.

**Comment by atiselsts** [comment]
Let's say that I could pull the rest, but this one I'm particularly feel uncomfortable about: https://github.com/atiselsts/contiki-ng/pull/3/commits/d44aa3eceac4e6a6b9e654555b3a32012f75ac66

Could you submit it in a separate PR? I mean, directly to Contiki-NG.

**Comment by atiselsts** [comment]
That "clean before build" option did not work at all. The `-j` flag is at fault - looks like that if given multiple targets, `make` tries to build them concurrently. So we end up in a situation when the compilation is started before the cleaning is finished.

If we do need to clean before build (not sure about that, but can't hurt), I propose adding this dependency in the Makefile `all: clean`.

**Comment by simonduq** [comment]
How about adding a separate call to `make clean`, without `-j`, prior to
building?

**Comment by simonduq** [comment]
I'll submit
https://github.com/atiselsts/contiki-ng/commit/d44aa3eceac4e6a6b9e654555b3a32012f75ac66
in a separate PR yes.

**Comment by atiselsts** [comment]
ok, fine. I updated to Contiki-NG PR with clean-before-build and will close this now.

**Comment by simonduq** [comment]
You mention in https://github.com/contiki-ng/contiki-ng/pull/311 that everything builds -- does this mean is no longer necessary?

**Comment by atiselsts** [comment]
That is correct - at least on my server. I assume you also changed some makefiles and some example-specific configuration.

**Pull request closed without merging**